test/spacesuit_cors_middleware_test.exs

```diff
@@ -0,0 +1,203 @@
+defmodule SpacesuitCorsMiddlewareTest do
+  use ExUnit.Case
+  doctest Spacesuit.CorsMiddleware
+
+  @original_app_config Application.get_env(:spacesuit, :cors)
+
+  setup_all do
+    req = %{
+      :headers => %{"origin" => "http://localhost"},
+      :scheme  => "http",
+      :host    => "www.example.com",
+      :port    => 80,
+      :path    => "/matched",
+      :method  => "GET"
+    }
+    {:ok, req: req}
+  end
+
+  describe "changing various configuration settings" do
+    test "passes all requests when the middleware is disabled", state do
+      # Override the :enabled setting
+      current = Application.get_env(:spacesuit, :cors)
+      Application.put_env(:spacesuit, :cors, Map.merge(current, %{enabled: false}))
+
+      # This would fail if the CORS support were enabled
+      req = Map.merge(state[:req], %{:headers => %{"origin" => ""}})
+      env = %{}
+      assert {:ok, ^req, ^env} = Spacesuit.CorsMiddleware.execute(req, env)
+
+      Application.put_env(:spacesuit, :cors, @original_app_config)
```

Comment:
could prolly move this line to a [`setup`](http://blog.lucidsimple.com/2016/01/31/exunit-cheat-sheet.html#callbacks) method that runs before each test and remove this here and elsewhere so as not to be forgotten; up to you